### PR TITLE
Use report_interval for the frequency of storing the energy for Gromacs process

### PR DESCRIPTION
**Pull request by xiki-tempula**
Fix the test where assign nsteps=0 might be duplicated.

### CODE REVIEW

File: test/Sandpit/Exscientia/Process/test_single_point_energy.py [hunk 1]
```diff
@@ -84,7 +98,13 @@ def test_amber_gromacs_triclinic(system):
 
     # Modify the GROMACS configuration to run zero steps.
     config = process_gmx.getConfig()
-    config[2] = "nsteps = 0"
```
Comment (xiki-tempula) [inline test/Sandpit/Exscientia/Process/test_single_point_energy.py:87]: Deal with the case that the nsteps is not the third term

File: python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py [hunk 2]
```diff
@@ -531,8 +533,12 @@ def tranform(charge, LJ):
...
                 "init-lambda-state"
             ] = self.protocol.getLambdaIndex()  # Current lambda value.
-            protocol_dict["nstcalcenergy"] = 200  # Calculate energies every 200 steps.
-            protocol_dict["nstdhdl"] = 200  # Write gradients every 200 steps.
+            protocol_dict[
+                "nstcalcenergy"
+            ] = self._report_interval  # Calculate energies every report_interval steps.
+            protocol_dict[
+                "nstdhdl"
+            ] = self._report_interval  # Write gradients every report_interval steps.
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py:541]: Don't the `BSS` defaults set this report interval to a very high value for the different protocols? Is this a good idea?

Comment (msuruzhon) [reply python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py:541]: Ah never mind, they are low now, I thought they used to be high...

**Review by msuruzhon — APPROVED** [review]
LGTM

**Pull request merged**